builtin/providers/google/resource_container_cluster.go

```diff
@@ -282,6 +288,24 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 		cluster.InitialClusterVersion = v.(string)
 	}
 
+	if v, ok := d.GetOk("additional_zones"); ok {
```

Comment:
Ah, you're right. The API docs claim you can update locations but the client library hasn't been updated with that change yet (hopefully that'll get fixed soon, I've been doing a fair amount of poking but it's apparently complicated). `ForceNew: True` sounds good then.